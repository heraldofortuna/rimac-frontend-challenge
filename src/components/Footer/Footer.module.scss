.footer {
  background: $grey-100-color;
  padding: 2rem;
  margin: auto 0 0 0;

  .container {
    max-width: 1200px;
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    gap: 2rem;
    margin: 0 auto;

    .logo {
      &--mobile {
        display: block;
        
        @media (min-width: $breakpoint-desktop) {
          display: none;
        }
      }

      &--desktop {
        display: none;
        
        @media (min-width: $breakpoint-desktop) {
          display: block;
        }
      }
    }

    .separator {
      height: 1px;
      width: 100%;
      background: $grey-80-color;

      @media (min-width: $breakpoint-desktop) {
        display: none;
      }
    }

    .text {
      color: $grey-10-color;
      font-size: 12px;

      @media (min-width: $breakpoint-desktop) {
        font-size: 14px;
      }
    }
  }

  @media (min-width: 768px) {
    .container {
      flex-direction: row;
    }
  }
}