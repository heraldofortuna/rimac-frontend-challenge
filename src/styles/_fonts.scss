@font-face {
  font-family: 'BR Sonoma Regular';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular'), url('/fonts/BRSonoma-Regular-BF654c45266c042.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Regular Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular Italic'), url('/fonts/BRSonoma-RegularItalic-BF654c452681c11.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin'), url('/fonts/BRSonoma-Thin-BF654c45255ffe0.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin Italic'), url('/fonts/BRSonoma-ThinItalic-BF654c45268d3f9.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight'), url('/fonts/BRSonoma-ExtraLight-BF654c45265af8d.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight Italic'), url('/fonts/BRSonoma-ExtraLightItalic-BF654c4525a5046.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light'), url('/fonts/BRSonoma-Light-BF654c452608e0f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light Italic'), url('/fonts/BRSonoma-LightItalic-BF654c45266aa83.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium'), url('/fonts/BRSonoma-Medium-BF654c45266edd1.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium Italic'), url('/fonts/BRSonoma-MediumItalic-BF654c45267d45f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold'), url('/fonts/BRSonoma-SemiBold-BF654c45268c340.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold Italic'), url('/fonts/BRSonoma-SemiBoldItalic-BF654c452696350.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold'), url('/fonts/BRSonoma-Bold-BF654c4526823f5.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold Italic'), url('/fonts/BRSonoma-BoldItalic-BF654c4525c9c27.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black'), url('/fonts/BRSonoma-Black-BF654c4525506bf.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black Italic'), url('/fonts/BRSonoma-BlackItalic-BF654c45268988e.woff') format('woff');
}