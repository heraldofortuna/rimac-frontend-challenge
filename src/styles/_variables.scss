$grey-100-color: #03050F;
$grey-80-color: #2B304E;
$grey-30-color: #CCD1EE;
$grey-10-color: #F8F9FF;
$blue-berry-600-color: #4F4FFF;
$neutrals-700-color: #141938;
$neutrals-600-color: #7981B2;
$neutrals-0-color: #FFFFFF;

$sonoma-font: 'BR Sonoma Regular', sans-serif;
$lato-font: 'Lato', sans-serif;

$breakpoint-mobile: 480px;
$breakpoint-desktop: 768px;