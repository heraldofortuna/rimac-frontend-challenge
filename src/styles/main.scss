@use 'fonts';
@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700;900&display=swap');

* {
  padding: 0;
  margin: 0;
}

body {
  background-color: $grey-10-color;
  font-family: $sonoma-font;
  color: $grey-100-color;
}

#root {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}