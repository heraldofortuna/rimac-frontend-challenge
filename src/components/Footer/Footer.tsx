import RimacLogo from "@assets/logo-white.svg";
import RimacMobileLogo from "@assets/logo-mobile-white.svg";
import styles from "./Footer.module.scss";
import getCurrentYear from "../../utils/getCurrentYear/getCurrentYear";
const currentYear = getCurrentYear();

const Footer: React.FC = () => {
  return (
    <footer className={styles.footer}>
      <div className={styles.container}>
        <img
          src={RimacMobileLogo}
          alt="Mobile Rimac Logo"
          className={styles["logo--mobile"]}
        />
        <img
          src={RimacLogo}
          alt="Desktop Rimac Logo"
          className={styles["logo--desktop"]}
        />
        <span className={styles.separator}></span>
        <p className={styles.text}>Â© {currentYear} RIMAC Seguros y Reaseguros.</p>
      </div>
    </footer>
  );
};

export default Footer;
