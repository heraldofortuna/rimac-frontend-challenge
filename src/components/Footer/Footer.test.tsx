import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";
import Footer from "./Footer";
import styles from "./Footer.module.scss";
import getCurrentYear from "../../utils/getCurrentYear/getCurrentYear";

vi.mock("@assets/logo-white.svg", () => ({
  default: "rimac-logo-white.svg",
}));

vi.mock("@assets/logo-mobile-white.svg", () => ({
  default: "rimac-logo-mobile-white.svg",
}));

describe("Footer Component", () => {
  let container: HTMLElement;
  let footer: HTMLElement;

  beforeEach(() => {
    const { container: renderedContainer } = render(<Footer />);
    container = renderedContainer;
    footer = screen.getByRole("contentinfo");
  });

  it("renders the footer element with correct class", () => {
    expect(footer).toBeInTheDocument();
    expect(footer).toHaveClass(styles.footer);
  });

  it("renders the container div with correct class", () => {
    const containerDiv = footer.firstElementChild;
    expect(containerDiv).toHaveClass(styles.container);
  });

  it("renders the mobile logo with correct attributes", () => {
    const mobileLogo = screen.getByAltText("Mobile Rimac Logo");
    expect(mobileLogo).toBeInTheDocument();
    expect(mobileLogo).toHaveAttribute("src", "rimac-logo-mobile-white.svg");
    expect(mobileLogo).toHaveClass(styles["logo--mobile"]);
  });

  it("renders the desktop logo with correct attributes", () => {
    const desktopLogo = screen.getByAltText("Desktop Rimac Logo");
    expect(desktopLogo).toBeInTheDocument();
    expect(desktopLogo).toHaveAttribute("src", "rimac-logo-white.svg");
    expect(desktopLogo).toHaveClass(styles["logo--desktop"]);
  });

  it("renders the separator", () => {
    const separator = container.querySelector(`.${styles.separator}`);
    expect(separator).toBeInTheDocument();
  });

  it("displays the correct copyright text", () => {
    const currentYear = getCurrentYear();
    const copyrightText = screen.getByText(
      `Â© ${currentYear} RIMAC Seguros y Reaseguros.`,
    );
    expect(copyrightText).toBeInTheDocument();
    expect(copyrightText).toHaveClass(styles.text);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
