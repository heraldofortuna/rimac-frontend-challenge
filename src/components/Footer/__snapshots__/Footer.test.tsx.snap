// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="_footer_0a5567"
  >
    <div
      class="_container_0a5567"
    >
      <img
        alt="Mobile Rimac Logo"
        class="_logo--mobile_0a5567"
        src="rimac-logo-mobile-white.svg"
      />
      <img
        alt="Desktop Rimac Logo"
        class="_logo--desktop_0a5567"
        src="rimac-logo-white.svg"
      />
      <span
        class="_separator_0a5567"
      />
      <p
        class="_text_0a5567"
      >
        Â© 2025 RIMAC Seguros y Reaseguros.
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
